src/components/StatsInnerList.tsx

```diff
@@ -0,0 +1,86 @@
+import * as React from "react";
+import { LibraryData } from "../interfaces";
+import { getPercentage } from "../utils/sharedFunctions";
+
+export interface StatsInnerListProps {
+  data: any;
+  styled: boolean;
+  stagesToShow?: {[ key: string ]: boolean};
+  showGeographicInfo?: boolean;
+}
+
+export default class StatsInnerList extends React.Component<StatsInnerListProps, {}> {
+  makeCategoryBar(category: string): JSX.Element[] {
+    let catName = category.replace(category[0], category[0].toUpperCase());
+    let length = this.props.data[category].length;
+    let allLengths = Object.values(this.props.data).map(x => (x as any).length);
```

Comment:
Not a big deal, but you're calculating this allLengths every time you make a new `li`. So maybe it could be done in `render` to do it once. The only drawback is that now you're passing in a value down two functions, and maybe the trade-off isn't bad.